channeld/full_channel.c

```diff
@@ -298,6 +300,18 @@ static enum channel_add_err add_htlc(struct channel *channel,
 	const struct channel_view *view;
 	size_t i;
 
+	/* BOLT #2:
+	 *
+	 * A receiving node MUST fail the channel if:
+	 *...
+	 * - the chain_hash value is set to a hash of a chain that is unknown
+	 *   to the receiver.
+     */
+	const struct chainparams *chain_params = chainparams_by_hash(chain_hash);
```

Comment:
I don't understand why this is being checked here at all. The `chain_hash` is channel-specific, and the chain is implicitly communicated through the channel that is being updated, so we should not be passing this down here at all.